ETLParser.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+
+# Parser for PEDSnet ETL conventions document
+
+import json
+import os.path
+import re
+import sys
+
+_h1 = re.compile(r'^\#[^\#]+')
+_h2 = re.compile(r'^\#\#[^\#]+')
+
+
+class Document():
+    def __init__(self, path):
```

Comment:
For flexibility, this should take a file-like object (could be a file handler or a stream) and force the caller to handle closing the file or stream. This also enables you to consume the file line-by-line rather loading all the data into memory via `f.readlines()`. Any place you are consuming lines would just do:

```python
for line in self.fileobj:
    ...
```